### ISSUE TITLE
Keys Expire After First Post TTL Access

### ISSUE THREAD

**Issue by khobbs-ccri** [fppt/jedis-mock#122]
We just started adding TTLs to our library. The library uses Jedis, is written in Scala, and uses com.github.fppt.jedismock.RedisServer in unit tests.

We just learned that tests of our codes use of hashmap expiration are failing because the key exists in jedis-mock until first access.

The exploratory code we just added is :

```
          val jedis = ras.pool.getResource
          jedis.hset("test", "key", "value")
          println(jedis.expire("test", 1))
          println(jedis.keys("*"))
          Thread.sleep(3000)
          println(jedis.keys("*"))
          jedis.get("test")
          println(jedis.keys("*"))
          jedis.close

``` 
and that prints:

```
1
[test]
[test]
[]
```

We see different behavior when we run the tests against a Redis run locally. The key does not appear in the first scan after expiration.

**Comment by khobbs-ccri** [comment]
I'm told this may be related to https://github.com/redis/jedis/issues/2374

I realize I left off that we're using
redis.clients.jedis-3.5.2
com.github.fppt.jedis-mock-0.1.19

**Comment by khobbs-ccri** [comment]
Are our scans of keys using this block?
https://github.com/fppt/jedis-mock/blob/master/src/main/java/com/github/fppt/jedismock/storage/RedisBase.java#L23-L25
It does not seem to check the TTLs.

**Comment by wcedmisten** [comment]
Related, but I think Scan is also missing the TTL functionality. I can file this as a separate ticket if you want.

```
import com.github.fppt.jedismock.RedisServer;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.ScanParams;

RedisServer server = RedisServer.newRedisServer();
server.start();

Jedis jedis = new Jedis(server.getHost(), server.getBindPort());

jedis.hset("test", "key", "value");
jedis.expire("test", 1);

System.out.println("Before expiry:");
System.out.println(jedis.scan(ScanParams.SCAN_POINTER_START).getResult());

Thread.sleep(3000);

System.out.println("After expiry:");
System.out.println(jedis.scan(ScanParams.SCAN_POINTER_START).getResult());
```

With jedis-mock:
```
Before expiry:
[test]
After expiry:
[test]
```

With redis:
```
Before expiry:
[test]
After expiry:
[]
```

**Comment by inponomarev** [comment]
@wcedmisten this has been already fixed in #122 (expiration check is implemented on a low level). However, thank you for your notice, I added an extra test for this: https://github.com/fppt/jedis-mock/blob/a58ab3c452ae8d0f47b083e8af88133d6849c5af/src/test/java/com/github/fppt/jedismock/comparisontests/TestScan.java#L68

The fix will be included in the next release!
